iceoryx_binding_c/include/iceoryx_binding_c/publisher.h

```diff
@@ -24,20 +24,29 @@
 /// @brief publisher handle
 typedef struct cpp2c_Publisher* iox_pub_t;
 
+/// @brief options to be set for a publisher
+typedef struct
+{
+    // size of the history chunk queue
+    uint64_t historyCapacity;
+
+    // name of the node the publisher belongs to
+    const char* nodeName;
+} iox_publisher_options_t;
+
 /// @brief creates a publisher handle
 /// @param[in] self pointer to preallocated memory of size = sizeof(iox_pub_storage_t)
 /// @param[in] service serviceString
 /// @param[in] instance instanceString
 /// @param[in] event eventString
-/// @param[in] historyCapacity size of the history chunk queue
-/// @param[in] nodeName name of the node the publisher belongs to
+/// @param[in] options publisher options set by the user
 /// @return handle of the publisher
+/// @attention the options must be set to valid values
 iox_pub_t iox_pub_init(iox_pub_storage_t* self,
                        const char* const service,
                        const char* const instance,
                        const char* const event,
-                       const uint64_t historyCapacity,
-                       const char* const nodeName);
+                       const iox_publisher_options_t options);
```

Comment:
I always thought one cannot pass structs to C functions or is it only regarding returning structs?

Nevertheless, if a pointer is passed, it could be interpreted as using the default options. This would somehow mimic the C++ behavior.

Reply:
You can pass it by value or as pointer.

But If I understand you right you want nullptr to indicate default options? (It is not clear from what you wrote).